forte/data/extractor/attribute_extractor.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from collections import Hashable, abc
+from typing import Dict, Any, Union, Iterable
+from ft.onto.base_ontology import Entry, Annotation
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.data.converter.feature import Feature
+from forte.data.extractor.base_extractor import BaseExtractor
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "AttributeExtractor"
+]
+
+
+class AttributeExtractor(BaseExtractor):
+    r"""AttributeExtractor extracts feature from the attribute of entry.
+
+    Args:
+        config: An instance of `Dict` or
+            :class:`forte.common.configuration.Config`
+
+            attribute (str): Required. The attribute name of the
+                entry from which features will be extracted. For
+                example, "text" attribute of Token.
+    """
+    def __init__(self, config: Union[Dict, Config]):
+        super().__init__(config)
+
+        if "attribute" not in self.config:
+            raise AttributeError("attribute needs to be specified in "
+                                "the configuration of an AttributeExtractor.")
+
+    @property
+    def state(self) -> Dict:
+        state = super().state
+        state.update({
+            "attribute": self.config.attribute
+        })
+        return state
+
+    @classmethod
+    def from_state(cls, state: Dict) \
+        -> Union["BaseExtractor", "AttributeExtractor"]:
+        obj = BaseExtractor.from_state(state)
+        additional = {
+            "attribute": state["attribute"]
+        }
+        obj.config = Config(obj.config, additional,
+                            allow_new_hparam=True)
+        obj.__class__ = cls
+        return obj
+
+    @staticmethod
+    def get_attribute(entry: Entry, attr: str) -> Any:
```

Comment:
We may actually move these staticmethod to a util file, like here https://github.com/asyml/forte/blob/master/forte/data/data_utils.py